Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -1520,13 +1520,26 @@ signal_t MatrixWorkspace::getSignalAtCoord(
   }
 
   if (wi < nhist) {
-    const auto &X = this->binEdges(wi);
-    auto it = std::lower_bound(X.cbegin(), X.cend(), x);
-    if (it == X.end()) {
+    const auto &xVals = x(wi);
+    auto it = std::lower_bound(xVals.cbegin(), xVals.cend(), xCoord);
+    if (it == xVals.end()) {
+      if (!isHistogramData()) {
+        auto deltaX = (xVals[xVals.size() - 1] - xVals[xVals.size() - 2]) / 2.0;
```

Comment:
`vector.back()` would be useful here to avoid having to call `size()` all of the time. It's also quicker to multiply than to divide so it would be better to write `0.5*` than `/2.0`. How about something like:

```cpp
auto binEnd = xVals.back() + 0.5*(xVals.back() - *std::next(xVals.rbegin()));
if(xCoord < binEnd)
  return yVals.back()
```